src/common/nvtx.h

```diff
@@ -0,0 +1,104 @@
+/*
```

Comment:
Should this new file be ./src/common/cuda/nvtx.h ?

Reply:
Could be, will change that.